src/ImmersedBoundaries/immersed_reductions.jl

```diff
@@ -21,6 +21,13 @@ const IF = AbstractField{<:Any, <:Any, <:Any, <:ImmersedBoundaryGrid}
 @inline condition_operand(func::Function,         op::IF, ::Nothing, mask) = ConditionalOperation(op; func, condition=NotImmersed(truefunc), mask)
 @inline condition_operand(func::typeof(identity), op::IF, ::Nothing, mask) = ConditionalOperation(op; func, condition=NotImmersed(truefunc), mask)
 
+@inline function condition_operand(func::Function, op::IF, cond::AbstractArray, mask)
+    arch = architecture(op.grid)
+    arch_condition = arch_array(arch, cond)
```

Comment:
It would be nice if we make this work also for fields, which are `AbstractArrays`, but do not have an `arch_array` method defined

Reply:
something like
```suggestion
    arch_condition = arch_array(arch, parent(cond))
```
might suffice

Reply:
`on_architecture` was meant for this

Reply:
> something like
> 
> might suffice

I don't like that. Why not keep it as a Field?